SimTKcommon/Scalar/include/SimTKcommon/internal/NTraits.h

```diff
@@ -799,9 +799,14 @@ template <class R> class NTraits< conjugate<R> > {
     static TNormalize normalize(const T& t) {return TNormalize(t/abs(t));}
 
     // 1/conj(z) = conj(1/z), for complex z.
-    static TInvert invert(const T& t)    
-    {   const typename NTraits<THerm>::TInvert cmplx(NTraits<THerm>::invert(t.conj()));
-        return reinterpret_cast<const TInvert&>(cmplx); } // recast complex to conjugate it
+    // Here reinterpret_cast is source of [-Wstrict-aliasing] warning.
+    // New code is more explicit and should be faster(???).
+    static TInvert invert(const T& t){
+      return static_cast<TInvert>(TReal(1)/t.conj());
```

Comment:
Why do you think there would be bugs? The whole point of this very old piece of code was to get the optimizer to produce great code that executes quickly. That is one of the best reasons to use C++ over safer languages like Java, and is why things like `reinterpret_cast` are in the language. In this case the warning is presumably that using `reinterpret_cast` is dangerous, which it is. But that isn't the same as saying it shouldn't be used.

You shouldn't touch this code if you don't fully understand it and have regression tests to make sure it still works correctly. What you could do is locally suppress the warning here so that you could still run `-Wall`. Please note that we build `-Werror` so our CI builds fail if there is even a single warning issued anywhere. If you do suppress the warning please do it over as small an are as possible and provide clear documentation about what you're doing and why.